lerobot-web-fe/src/services/status.service.ts

```diff
@@ -9,27 +11,33 @@ interface StatusResponse {
   is_recording: boolean;
   leader_follower_status: boolean;
   name: string;
-  robot_status: { device_name: string; name: string; robot_type: string }[];
+  robot_status: { device_name: string | null; name: string; robot_type: string }[];
   robots: string[];
   server_ip: string;
   status: string;
   version_id: string;
 }
 
-export const API_BASE = 'http://127.0.0.1:80';
-
 export async function getStatus(): Promise<StatusResponse> {
-  const res = await fetch(`${API_BASE}/status`, {
-    method: 'GET',
-    headers: {
-      'Accept': 'application/json',
-      'Content-Type': 'application/json',
-    },
-  });
+  const { apiUrl } = useStatusStore.getState();
```

Comment:
Instead of destruct, please use the separated selector-based solution here as well.

```suggestion
  const apiUrl = useStatusStore((s) => s.apiUrl);
```

Reply:
It will cause an invalid hook call. The useStatusStore(s => s.apiUrl) is for reactive, efficient updates. The  useStatusStore.getState().apiUrl usage is safe, static, and meets React's hook rules.